libraries/default.rb

```diff
@@ -0,0 +1,67 @@
+def collect_github_repositories(token = {}, org)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
+
+  require 'octokit'
+
+  github = Octokit::Client.new(token)
+  github.auto_paginate = true
+
+  github.org_repos(org).map(&:name)
+end
+
+def set_up_github_push(token = {}, orgname, reponame, jobname, trigger_token)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
+
+  require 'octokit'
+  require 'pp'
+
+  github = Octokit::Client.new(token)
+
+  repopath = "#{orgname}/#{reponame}"
+  hooks = github.hooks(repopath)
+  type = 'web'
+  events = ['issue_comment']
+  content_type = 'form'
+  url = "https://#{node['jenkins']['master']['host']}" \
+    ":#{node['jenkins']['master']['port']}/job/#{jobname}/" \
+    "buildWithParameters?token=#{trigger_token}"
+
+  # Get all hooks that we may have created in the past
+  existing_hooks = hooks.select do |h|
+    h['name'] == type &&
+    h['events'] == events &&
+    h['config']['content_type'] == content_type
+  end
+
+  # If the hook exists and is outdated, delete it.
+  current_hook_exists = false
+  existing_hooks.each do |h|
+    if h['url'] != url
+      github.delete_hook(repopath, h['id'])
```

Comment:
I can't think of an elegant way to solve this problem since there doesn't seem to be a way to define a "comment" or anything for a hook to identify it as the cookbook uploader one.  I imagined the chances of there other hooks of type `web`, content type `form`, and only pushing `issue_comment` events would be pretty low.  I suppose a more foolproof option would be storing the hook id of the hook as an attribute on the node when it's created, but that also would accumulate hooks if the jenkins node ever got deleted and recreated since the hook ids would then be lost.